### PR TITLE
Implement Log Masking

### LINKED ISSUE #1720: Mask Sensitive Data in Logs
### Summary

When logging, we should not expose any personally identifiable information (PII) in logs.

### Value

Logs do not contain sensitive data.

### Specification

- [ ] Add a launcher flag that disables masking
- [ ] Mask user code and environment variables in logs

### Acceptance Criteria & Test Cases

- Logs don't contain sensitive user data.

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #1720

Changelog:

- feat: add `ToMaskedString` interface that indicates that the object has a custom string representation masking some PII. 
- feat: add `--no-log-masking` CLI flag that disables log masking

### Important Notes

What is masked:
- Literals containing code
- Values of executed expressions
- Values of the user environment variables
- Paths inside the user project

What is **not** masked:
- Names of functions and modules
- Keys (names) of the user environment variables
- System and distribution paths, including a path to the user project

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/language-server/src/main/scala/org/enso/languageserver/requesthandler/file/CreateFileHandler.scala
Comment (iamrecursion) [inline engine/language-server/src/main/scala/org/enso/languageserver/requesthandler/file/CreateFileHandler.scala]: Could the exception message contain sensitive data? The same question for the other exception messages we log.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala
Comment (iamrecursion) [inline engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:122]: Are you sure? I thought `getName` returned only the top name in the path, where `getPath` was the fully-qualified module name. We want the fully-qualified path.

**Review by iamrecursion — CHANGES_REQUESTED** [review]
Please document this whole thing in the docs. It needs to be somewhere that users can find.

Additionally, once this is merged please get a variety of people to test it and send the logs. We want to _check_ that we've not missed masking anything important.

I also think that any component that is running with log masking disabled should warn in the console.

File: engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala
Comment (4e6) [reply engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala:122]: It's not like that. `getName` returns qualified name, `getPath` returns the path to the source file

File: engine/language-server/src/main/scala/org/enso/languageserver/requesthandler/file/CreateFileHandler.scala
Comment (4e6) [reply engine/language-server/src/main/scala/org/enso/languageserver/requesthandler/file/CreateFileHandler.scala]: Right, IO errors can potentially leak paths inside the user project. I'll mask them as well

**Review by iamrecursion — APPROVED** [review]
Approved, conditional on moving the logging docs into `logging.md`.

**Pull request merged**
